honeybadgermpc/hbavss.py

```diff
@@ -182,3 +191,146 @@ def _get_dealer_msg(self, value):
             v = None if values is None else values[i]
             avss_tasks[i] = asyncio.create_task(self.avss(k*avss_id+i, v, dealer_id))
         return await asyncio.gather(*avss_tasks)
+
+
+class HbAvssBatch(HbAvss):
+    async def _process_avss_msg(self, avss_id, dealer_id, dispersal_msg):
+        tag = f"{dealer_id}-{avss_id}-AVSS"
+        send, recv = self.get_send(tag), self.subscribe_recv(tag)
+
+        def multicast(msg):
+            for i in range(self.n):
+                send(i, msg)
+
+        commitments, ephemeral_public_key, encrypted_witnesses = loads(
+            dispersal_msg)
+        secret_size = len(commitments)
+
+        # all_encrypted_witnesses: n
+        shared_key = pow(ephemeral_public_key, self.private_key)
+
+        shares = [None] * secret_size
+        witnesses = [None] * secret_size
+        # Decrypt
+        for k in range(secret_size):
+            shares[k], witnesses[k] = SymmetricCrypto.decrypt(
+                str(shared_key).encode("utf-8"),
+                encrypted_witnesses[self.my_id * secret_size + k])
+
+        # verify & send all ok
+        for i in range(secret_size):
+            if not self.poly_commit.verify_eval(
+                    commitments[i], self.my_id+1, shares[i], witnesses[i]):
+                # will be replaced by sending out IMPLICATE message later
+                # multicast(HbAVSSMessageType.IMPLICATE)
+                logging.error("PolyCommit verification failed.")
+                raise HoneyBadgerMPCError("PolyCommit verification failed.")
+
+        multicast(HbAVSSMessageType.OK)
+
+        # Bracha-style agreement
+        oks_recvd = 0
```

Comment:
you don't need these two variables if you are using a `set`, just call `len(ok_set)`?

Reply:
Resolved.